code/game/objects/items/carp_lasso.dm

```diff
@@ -1,59 +1,79 @@
 /obj/item/mob_lasso
 	name = "space lasso"
-	desc = "Comes standard with every space-cowboy.\nCan be used to tame space carp."
+	desc = "Comes standard with every space-cowboy.\n<span class='notice'>Can be used to tame space carp.</span>"
 	icon = 'icons/obj/carp_lasso.dmi'
 	icon_state = "lasso"
 	///Ref to timer
 	var/timer
 	///Ref to lasso'd carp
 	var/mob/living/simple_animal/mob_target
+	///Range we can lasso things at
+	var/range = 8
 	///Whitelist of allowed animals
-	var/list/whitelist_mobs = list(/mob/living/simple_animal/hostile/carp, /mob/living/simple_animal/cow, /mob/living/simple_animal/hostile/retaliate/dolphin)
+	var/list/whitelist_mobs
+	///blacklist of disallowed animals
+	var/list/blacklist_mobs
+
+/obj/item/mob_lasso/Initialize(mapload)
+	. = ..()
+	whitelist_mobs = typecacheof(list(/mob/living/simple_animal/hostile/carp, /mob/living/simple_animal/cow, /mob/living/simple_animal/hostile/retaliate/dolphin), only_root_path = TRUE)
```

Comment:
aren't there subtypes of carp

Reply:
Yes. That includes things like magicarps and chaos magicarps, though, which probably shouldn't be tameable with the base lasso. I can add the sharks to that list, though.